### PR TITLE
Support running reports with dependencies

### PR DESCRIPTION
**Pull request by richfitz**
Basic support for running orderly reports with dependencies, using the query DSL in https://github.com/mrc-ide/outpack/pull/15

~Requires https://github.com/mrc-ide/outpack/pull/18 and https://github.com/mrc-ide/outpack/pull/17~

The windows build failure is just gha failing to cope with gert being updated on CRAN, it will resolve in a day or two, but does not matter for this PR

### CODE REVIEW

File: tests/testthat/test-run.R [hunk 1]
```diff
@@ -154,3 +154,19 @@ test_that("clean parameters given spec", {
   expect_error(check_parameters(list(a = 1, c = 3, d = 4), info),
                "Extra parameters: 'c', 'd'")
 })
+
+
+test_that("can run packet with dependency", {
+  path <- test_prepare_orderly_example(c("minimal", "depend"))
+  env <- new.env()
+  id1 <- orderly_run("minimal", root = path, envir = env)
+  id2 <- orderly_run("depend", root = path, envir = env)
```
Comment (r-ash) [inline tests/testthat/test-run.R:163]: I tried running the `"depend"` task first and we get an error here but not the most transparent one. 

It errors with "Error in private$metadata_load(id) : id 'NA' not found in index" - worth adding a test for the error case and handling with a bit more of a useful message?

**Review by r-ash — CHANGES_REQUESTED** [review]
This looks great, just the one comment about error case which I think could give a more informative message.

File: tests/testthat/test-run.R (hunk 1 above)
Comment (richfitz) [reply tests/testthat/test-run.R:163]: definitely, thanks

File: R/orderly.R [hunk 2]
```diff
@@ -182,3 +203,35 @@ check_parameters <- function(parameters, info) {
...
+resolve_dependency <- function(name, query, parameters, root) {
+  ## TODO, pass location information through too, and join them into
+  ## the scope, but we don't yet really have orderly remotes
+  ## (vimc-6666)
+  scope <- bquote(name == .(name))
+  ## TODO: tell outpack we expect a single value (mrc-3493)
+  id <- outpack::outpack_query(query, parameters, scope,
+                               require_unpacked = TRUE, root = root$outpack)
+  if (is.na(id)) {
+    ## TODO: this friendlier to the user (mrc-3492)
```
Comment (hillalex) [inline R/orderly.R]: there's a duplicate `if` statement here

**Review by hillalex — APPROVED** [review]

**Review by r-ash — APPROVED** [review]

**Pull request merged**